imgui.go

```diff
@@ -1227,6 +1279,56 @@ func ColumnsCount() int {
 	return int(C.iggGetColumnsCount())
 }
 
+// BeginDragDropSource opens the scope for current draw and drop source.
+// Call when current ID is active.
+// When this returns true you need to: a) call SetDragDropPayload() exactly once, b) you may render the payload visual/description, c) call EndDragDropSource()
+func BeginDragDropSource(flags int) bool {
+	return C.iggBeginDragDropSource(C.int(flags)) != 0
+}
+
+// SetDragDropPayload sets the payload for current draw and drop source.
+// type is a user defined string of maximum 32 characters.
+// Strings starting with '_' are reserved for dear imgui internal types.
+// Data is copied and held by imgui.
+func SetDragDropPayload(type_ string, payload string, cond Condition) bool {
+	typeArg, typeFin := wrapString(type_)
+	defer typeFin()
+	out := payload + "\000" + strings.Repeat(" ", 512-len(payload))
```

Comment:
This is extra magic that misses explanation and I suspect this is type-specific. If the `data` is handled as arbitrary byte array, this extra magic goes to the caller.
This also includes the constant of 512 as the internal argument. Please rework.